tests/app/legacy/v2/notifications/test_rest.py

```diff
@@ -66,441 +72,318 @@ class TestPush:
 
-        current_timestamp = int(time.time())
-        payload: JWTPayloadDict = {
-            'iss': str(service.id),
-            'iat': current_timestamp,
-            'exp': current_timestamp + 60,
-        }
-        token = generate_token(sig_key=secret, payload=payload)
-        client = client_factory(token)
+        yield _wrapper
+        ...
 
-        with (
-            patch('app.auth.LegacyServiceDao.get_service', new=AsyncMock(return_value=service)),
-            patch('app.auth.LegacyApiKeysDao.get_api_keys', new=AsyncMock(return_value=[api_key])),
-            patch('app.legacy.v2.notifications.rest.validate_template', return_value=None),
-        ):
-            response = client.post(route, json=request_payload)
+    @pytest.fixture
+    def path_request(
+        self,
+    ) -> Generator[
+        Callable[
+            [
+                UUID,
+                str | None,
+                UUID | None,
+                str | None,
+                RecipientIdentifierModel | None,
+                dict[str, str | int | float | list[str | int | float] | PersonalisationFileObject] | None,
+            ],
+            dict[str, object],
+        ]
+    ]:
+        """Generator to build request data.
 
-        assert response.status_code == status.HTTP_201_CREATED
+        Yields:
+            Generator[ Callable[ [ str | None, UUID4 | None, str | None, RecipientIdentifierModel | None, dict[str, str | int | float | list[str | int | float] | PersonalisationFileObject] | None, ], dict[str, object], ], None, ]: Request data
+        """
 
-    @pytest.mark.parametrize('route', routes)
-    async def test_router_returns_403_wrong_admin_secret(
+        def _wrapper(
+            template_id: UUID4,
+            reference: str | None = None,
+            sms_sender_id: UUID4 | None = None,
+            phone_number: str | None = None,
+            recipient_identifier: RecipientIdentifierModel | None = None,
+            personalization: dict[str, str | int | float | list[str | int | float] | PersonalisationFileObject]
+            | None = None,
+        ) -> dict[str, object]:
+            # Default to the simplest. Template requires personalization
+            request_data: V2PostSmsRequestModel | V2PostNotificationRequestModel
+            if phone_number:
+                request_data = V2PostSmsRequestModel(
+                    phone_number=ValidatedPhoneNumber(phone_number),
+                    template_id=template_id,
+                    personalisation=personalization or {'verify_code': '12345'},
+                )
+            else:
+                request_data = V2PostNotificationRequestModel(
+                    recipient_identifier=recipient_identifier,
+                    template_id=template_id,
+                    personalisation=personalization or {'verify_code': '12345'},
+                )
+            # TODO: KWM Implement tests that use these before up for PR
```

Comment:
Creating a ticket for this after talking with @EvanParish. Will update with the ticket number.

Reply:
@k-macmillan Looks like this TODO is still here. Did you finish this?

Reply:
I just pushed a change that [tests the reference](https://github.com/department-of-veterans-affairs/va-enp-api/pull/262/commits/1c355a134b4e2ec1e9c9b218f3d4a8478b7c8450#diff-552c9bfc894aed1a836a1693c2e150ab3a8fd2ae49974956e860092fa6e8d836R289) use and I change the TODO to reference the new ticket, 272.